src/core/SampleBuffer.cpp

```diff
@@ -293,123 +217,112 @@ void SampleBuffer::update( bool _keep_settings )
 	}
 }
 
-
-void SampleBuffer::convertIntToFloat ( int_sample_t * & _ibuf, f_cnt_t _frames, int _channels)
+SampleBuffer::DataVector SampleBuffer::convertIntToFloat (int_sample_t * & _ibuf, f_cnt_t _frames, int _channels)
 {
 	// following code transforms int-samples into
 	// float-samples and does amplifying & reversing
 	const float fac = 1 / OUTPUT_SAMPLE_MULTIPLIER;
-	m_data = MM_ALLOC( sampleFrame, _frames );
+	DataVector vector(_frames);
 	const int ch = ( _channels > 1 ) ? 1 : 0;
 
-	// if reversing is on, we also reverse when
-	// scaling
-	if( m_reversed )
-	{
-		int idx = ( _frames - 1 ) * _channels;
-		for( f_cnt_t frame = 0; frame < _frames;
-						++frame )
-		{
-			m_data[frame][0] = _ibuf[idx+0] * fac;
-			m_data[frame][1] = _ibuf[idx+ch] * fac;
-			idx -= _channels;
-		}
-	}
-	else
+	int idx = 0;
+	for( f_cnt_t frame = 0; frame < _frames;
+					++frame )
 	{
-		int idx = 0;
-		for( f_cnt_t frame = 0; frame < _frames;
-						++frame )
-		{
-			m_data[frame][0] = _ibuf[idx+0] * fac;
-			m_data[frame][1] = _ibuf[idx+ch] * fac;
-			idx += _channels;
-		}
+		vector[frame][0] = _ibuf[idx+0] * fac;
+		vector[frame][1] = _ibuf[idx+ch] * fac;
+		idx += _channels;
 	}
 
 	delete[] _ibuf;
-}
 
-void SampleBuffer::directFloatWrite ( sample_t * & _fbuf, f_cnt_t _frames, int _channels)
-
-{
-
-	m_data = MM_ALLOC( sampleFrame, _frames );
-	const int ch = ( _channels > 1 ) ? 1 : 0;
-
-	// if reversing is on, we also reverse when
-	// scaling
-	if( m_reversed )
-	{
-		int idx = ( _frames - 1 ) * _channels;
-		for( f_cnt_t frame = 0; frame < _frames;
-						++frame )
-		{
-			m_data[frame][0] = _fbuf[idx+0];
-			m_data[frame][1] = _fbuf[idx+ch];
-			idx -= _channels;
-		}
-	}
-	else
-	{
-		int idx = 0;
-		for( f_cnt_t frame = 0; frame < _frames;
-						++frame )
-		{
-			m_data[frame][0] = _fbuf[idx+0];
-			m_data[frame][1] = _fbuf[idx+ch];
-			idx += _channels;
-		}
-	}
-
-	delete[] _fbuf;
+	return vector;
 }
 
 
 void SampleBuffer::normalizeSampleRate( const sample_rate_t _src_sr,
 							bool _keep_settings )
 {
 	// do samplerate-conversion to our default-samplerate
-	if( _src_sr != Engine::mixer()->baseSampleRate() )
+	if( _src_sr != sampleRate ())
 	{
-		SampleBuffer * resampled = resample( _src_sr,
-					Engine::mixer()->baseSampleRate() );
-		MM_FREE( m_data );
-		m_frames = resampled->frames();
-		m_data = MM_ALLOC( sampleFrame, m_frames );
-		memcpy( m_data, resampled->data(), m_frames *
-							sizeof( sampleFrame ) );
-		delete resampled;
+		m_data = resampleData (m_data, sampleRate (), sampleRate ());
```

Comment:
I think the second argument should be `_src_sr`.

Reply:
good catch! thanks!